### ISSUE TITLE
[Doc]: legend guide should be OO

### ISSUE THREAD

**Issue by story645** [matplotlib/matplotlib#20754]
### Documentation Link

https://matplotlib.org/devdocs/tutorials/intermediate/legend_guide.html

### Problem

The legend guide is all pyplot interface, which is especially discouraged in the types of examples (complicated, some needing state) in the guide. For example:
```python
from matplotlib.legend_handler import HandlerPatch


class HandlerEllipse(HandlerPatch):
    def create_artists(self, legend, orig_handle,
                       xdescent, ydescent, width, height, fontsize, trans):
        center = 0.5 * width - 0.5 * xdescent, 0.5 * height - 0.5 * ydescent
        p = mpatches.Ellipse(xy=center, width=width + xdescent,
                             height=height + ydescent)
        self.update_prop(p, orig_handle, legend)
        p.set_transform(trans)
        return [p]


c = mpatches.Circle((0.5, 0.5), 0.25, facecolor="green",
                    edgecolor="red", linewidth=3)
plt.gca().add_patch(c)

plt.legend([c], ["An ellipse, not a rectangle"],
           handler_map={mpatches.Circle: HandlerEllipse()})
```

### Suggested improvement

switch the tutorial to OO

### Matplotlib Version

3.4.2.post1539+gf8f693922c

### Matplotlib documentation version

3.4.2.post1539+gf8f693922c

**Comment by BhaswatiRoy** [comment]
Under legend location can we just add the "best" parameter with a sample of code with output since it is the most commonly used and most important. Also, we can add plots for some more legend locations to make things clearer. Just like the following ss->

![Screenshot 2021-07-30 132412](https://user-images.githubusercontent.com/78029145/127620511-957c01eb-ae0e-4b8e-ac4b-7bf7b4942354.jpg)

**Comment by timhoffm** [comment]
Closed by #20792.
